internal/command/test_test.go

```diff
@@ -1132,72 +1132,80 @@ Condition expression could not be evaluated at this time.
 }
 
 func TestTest_LocalVariables(t *testing.T) {
-	td := t.TempDir()
-	testCopyDir(t, testFixturePath(path.Join("test", "pass_with_local_variable")), td)
-	defer testChdir(t, td)()
-
-	provider := testing_command.NewProvider(nil)
-
-	providerSource, close := newMockProviderSource(t, map[string][]string{
-		"test": {"1.0.0"},
-	})
-	defer close()
-
-	streams, done := terminal.StreamsForTesting(t)
-	view := views.NewView(streams)
-	ui := new(cli.MockUi)
-
-	meta := Meta{
-		testingOverrides: metaOverridesForProvider(provider.Provider),
-		Ui:               ui,
-		View:             view,
-		Streams:          streams,
-		ProviderSource:   providerSource,
-	}
-
-	init := &InitCommand{
-		Meta: meta,
-	}
+	tcs := map[string]struct {
+		expected string
+		code     int
+		skip     bool
+	}{
+		"pass_with_local_variable": {
+			expected: `tests/test.tftest.hcl... pass
+  run "first"... pass
+  run "second"... pass
 
-	if code := init.Run(nil); code != 0 {
-		t.Fatalf("expected status code 0 but got %d: %s", code, ui.ErrorWriter)
+Success! 2 passed, 0 failed.
+`,
+			code: 0,
+		},
+		"pass_var_inside_variables": {
+			expected: "main.tftest.hcl... pass\n  run \"first\"... pass\n\nSuccess! 1 passed, 0 failed.\n",
+			code:     0,
+		},
 	}
 
-	c := &TestCommand{
-		Meta: meta,
-	}
-	code := c.Run([]string{"-verbose", "-no-color"})
-	output := done(t)
+	for name, tc := range tcs {
+		t.Run(name, func(t *testing.T) {
+			if tc.skip {
+				t.Skip()
+			}
 
-	if code != 0 {
-		t.Errorf("expected status code 0 but got %d", code)
-	}
+			file := name
 
-	expected := `tests/test.tftest.hcl... pass
-  run "first"... pass
+			td := t.TempDir()
+			testCopyDir(t, testFixturePath(path.Join("test", file)), td)
+			defer testChdir(t, td)()
 
+			provider := testing_command.NewProvider(nil)
+			providerSource, providerClose := newMockProviderSource(t, map[string][]string{
+				"test": {"1.0.0"},
+			})
+			defer providerClose()
 
-Outputs:
+			streams, done := terminal.StreamsForTesting(t)
+			view := views.NewView(streams)
+			ui := new(cli.MockUi)
+			meta := Meta{
+				testingOverrides: metaOverridesForProvider(provider.Provider),
+				Ui:               ui,
+				View:             view,
+				Streams:          streams,
+				ProviderSource:   providerSource,
+			}
 
-foo = "bar"
-  run "second"... pass
+			init := &InitCommand{
+				Meta: meta,
+			}
 
-No changes. Your infrastructure matches the configuration.
+			if code := init.Run(nil); code != 0 {
+				t.Fatalf("expected status code 0 but got %d: %s", code, ui.ErrorWriter)
+			}
 
-OpenTofu has compared your real infrastructure against your configuration and
-found no differences, so no changes are needed.
+			command := &TestCommand{
+				Meta: meta,
+			}
 
-Success! 2 passed, 0 failed.
-`
```

Comment:
Any reason why the current "expected" value of the `pass_with_local_variable` test has been reduced and does not contain the entire output, as it used to before this PR?